### ISSUE TITLE
readpreference seems not working

### ISSUE THREAD

**Issue by no7dw** [balderdashy/sails-mongo#291]
I wanna set readpreference to my connection in my project which I'm using sails to connection with mongodb.
However, after checking my mongodb.log,  I still still see connection & query has been made to connection with primary  instead of secondary .
I'm using :

    sails V0.10.5
    sails-mongo:V0.10.4
    node:V0.10.35

here is my config in connection.js

    prodMongodb: {
        module: 'sails-mongo',
        host:  'pri.local',
        port:  27017,
        user: 'username',
        password: 'password',
        database: 'dbname',
        readPreference:"secondary",
        // Replica Set (optional)
        replSet: {
          servers: [
            {
              host: 'pri.local',
              port: 27017 // Will override port from default config (optional)
            },
            {
              host: 'sec1.local',
              port: 27017
            },
            {
              host: 'sec2.local',
              port: 27017
            }
          ],
          // See http://mongodb.github.io/node-mongodb-native/api-generated/replset.html (optional)
          options: {

            poolSize: 1,
            reconnectWait: 5 * 60 * 1000,
            retries: 10,
            socketOptions: {
              keepAlive: 0,
              connectTimeoutMS: 5 * 60 * 1000,
              socketTimeoutMS: 5 * 60 * 1000
            }
          }
        }
      }

**Comment by devinivy** [comment]
The connection options are built [here](https://github.com/balderdashy/sails-mongo/blob/bd5d0b1cd4fcb66f007d2eb242e605b2e5f9db66/lib/connection.js#L113).  Is `readPreference` used correctly there?

**Comment by no7dw** [comment]
@devinivy I've read the code ,and console.log that config ,and see the readPreference:"secondary" has been set .
but I log out mongodb.log , still no query was sent to secondary

**Comment by devinivy** [comment]
I see that `readPreference` should not be a string: http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html

**Comment by leoliew** [comment]
@devinivy readPreference  can use  a string value ,  Just in a different form mongo native,   please see this source:  https://github.com/mongodb/node-mongodb-native/blob/2.0/lib/read_preference.js  ,   is a enum type in mongo native .

@no7dw use replSet options in sails have two ways : 
1.using url parameter , like this 
mongo : {
    adapter: 'sails-mongo',
    url:"mongodb://host1:27017,host2:27017,host3:27017/sailsdemo?replicaSet=rs_name"
}

2.convert options to mongo url by mongo's native:
i see this document http://mongodb.github.io/node-mongodb-native/driver-articles/mongoclient.html?highlight=readpreference
the readPreference options may be in replSet. options object , like this 

replSet: {
      servers: [
        {
          host: 'pri.local',
          port: 27017 // Will override port from default config (optional)
        },
        {
          host: 'sec1.local',
          port: 27017
        },
        {
          host: 'sec2.local',
          port: 27017
        }
      ],
      // See http://mongodb.github.io/node-mongodb-native/api-generated/replset.html (optional)
      options: {
        readPreference:"secondary",
        poolSize: 1,
        reconnectWait: 5 * 60 * 1000,
        retries: 10,
        socketOptions: {
          keepAlive: 0,
          connectTimeoutMS: 5 * 60 * 1000,
          socketTimeoutMS: 5 * 60 * 1000
        }
      }
 
  I don't have a replicaset to test on right now , you should try two mentioned above

**Comment by no7dw** [comment]
@devinivy https://github.com/mongodb/node-mongodb-native/blob/2.0/lib/read_preference.js here it use string type.
finally I use the following instead

    url:"mongodb://host1:27017,host2:27017,host3:27017/sailsdemo?readPreference=secondary"

**Comment by sailsbot** [comment]
Thanks for posting, @no7dw.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
